load_balancer/status.js

```diff
@@ -0,0 +1,75 @@
+/* eslint import/no-dynamic-require: 0 */
+const { Logger } = require('../util/logger.js');
+const { check } = require('../util/environmentCheck.js');
+const request = require('request-promise-native');
+
+/* The environment variable LBCONFIG will contain the path to the config file.
+  The actual path for the config is "../deploy/configs/load_balancer/nodes_data_conf.json"
+  For the docker containers, the path is /etc/load_balancer/nodes_data_conf.json */
+check('LBCONFIG');
+
+const lbConfig = require(process.env.LBCONFIG);
+
+const loggerConfig = lbConfig.load_balancer;
+loggerConfig.cmd = 'log';
+
+const logger = new Logger(loggerConfig);
+
+async function getComponentStatus(node) {
```

Comment:
How about the following code structure? (BTW, there is a compilation error in the code snippet, please only use the idea, not the actual code)

```javascript
function isValidNode(node) {
        hostname = node.hostname;
        port = node.port;
        role = node.role;

        if (hostname === undefined || port === undefined || role == undefined)
                return false;
        else
                return true;
}

async function getComponentStatus(node) {
        if (!isValidNode(node))
                throw new InvalidNodeException();

        const resultJson = Object.assign({}, node);
        try {
                const body = await request.get(`https://${node.hostname}:${node.port}/connectionCheck`);
                if (body.toString() === 'true') {
                        logger.debug(`Node at ${node.hostname}:${node.port} is up and running.`);
                resultJson.status = 'up';
            }
        } catch (err) {
                logger.error(`Error connecting to ${node.hostname}:${node.port}`);
                logger.error(err);
                resultJson.status = 'down';
}
```

You also need to define InvalidNodeException class. I see the value in putting both isNodeValid() and InvalidNodeException in a module inside util/. That way, other components can cross check their configuration values as well.

Reply:
What properties wil the error class would require apart from an error message?

Reply:
just the error status message is sufficient.